src/configure/configure.ts

```diff
@@ -648,14 +656,69 @@ class Orchestrator {
             throw error;
         }
     }
+
+    private createProvisioningConfigurationObject(templateId: string, branch: string, pipeplineTemplateParameters: { [key: string]: string }, mode: provisioningMode): ProvisioningConfiguration {
+        return {
+          id: null,
+          branch,
+          pipelineTemplateId: templateId,
+          pipelineTemplateParameters: pipeplineTemplateParameters,
+          provisioningMode: mode,
+        } as ProvisioningConfiguration;
+      }
+
+    private async configurePipelineRemotely(): Promise<void>{
+        const provisioningConfigurer = new ProvisioningConfigurer(this.localGitRepoHelper);
+        const template = this.inputs.pipelineConfiguration.template as RemotePipelineTemplate;
+        try {
+            // prerequisite params
+            telemetryHelper.setCurrentStep('ConfiguringPreRequisiteParams');
+            await provisioningConfigurer.CreatePreRequisiteParams(this.inputs);
+
+            // Draft pipeline step
+            telemetryHelper.setCurrentStep('ConfiguringDraftPipeline');
+            let provisioningServiceDraftModeResponse: ProvisioningConfiguration;
```

Comment:
should we move this job queuing and waiting part to the configurer? This way the current function can be simpler.

Reply:
A counterargument could be that we want to treat queueing and waiting as two separate steps. In that case, you can keep the logic here, but then you should log setCurrentStep for both separately.

Reply:
Added two different. for Complete case we are logging steps in postSteps function

Reply:
Nice

Reply:
You can still move this code till L687 to the configurer to keep current function small.

Reply:
Thanks for suggesting this changes. This change has made this function quite simple.